slsim/Sources/SupernovaeCatalog/supernovae_sample.py

```diff
@@ -78,56 +84,63 @@ def supernovae_catalog(self, redshift=None, host_galaxy=True):
+                    "redshift should be None while host_galaxy is True."
+                    " Please set redshift to be None."
+                )
         else:
             host_galaxies = None
+            if redshift is not None:
+                supernovae_redshift = redshift
+            else:
+                raise ValueError(
+                    "redshift should be provided while host_galaxy is not True."
+                    " Please provide redshift list."
+                )
         time = []
         # Initialize a list attribute for each band in self.band_list
         for band in self.band_list:
             setattr(self, f"magnitude_{band}", [])
-        if redshift is None:
-            if host_galaxy is True:
-                supernovae_redshift = host_galaxies["z"]
-            else:
-                raise ValueError(
-                    "host_galaxy should be True while redshift is None."
-                    "Either set host_galaxy to True or provide redshift list"
-                )
-        else:
-            supernovae_redshift = redshift
         # generate lightcurve for each supernovae.
-        for z in supernovae_redshift:
-            lightcurve_class = random_supernovae.RandomizedSupernova(
-                self.sn_type,
-                z,
-                self.absolute_mag,
-                self.absolute_mag_band,
-                self.mag_zpsys,
-                self.cosmo,
-            )
-            time.append(self.lightcurve_time)
-            for band in self.band_list:
-                mag = lightcurve_class.get_apparent_magnitude(
-                    self.lightcurve_time, "lsst" + band, zpsys=self.mag_zpsys
+        if lightcurve is True:
+            for z in supernovae_redshift:
+                lightcurve_class = random_supernovae.RandomizedSupernova(
+                    self.sn_type,
+                    z,
+                    self.absolute_mag,
+                    self.absolute_mag_band,
+                    self.mag_zpsys,
+                    self.cosmo,
+                    self.sn_modeldir,
                 )
-                getattr(self, f"magnitude_{band}").append(mag)
-        lightcurve_data = {"MJD": time}
-        for band in self.band_list:
-            lightcurve_data["ps_mag_" + band] = getattr(self, f"magnitude_{band}")
-        # astropy table of supernovae lightcurves
-        lightcurve_table = Table(lightcurve_data)
+                time.append(self.lightcurve_time)
+                for band in self.band_list:
+                    mag = lightcurve_class.get_apparent_magnitude(
+                        self.lightcurve_time, "lsst" + band, zpsys=self.mag_zpsys
+                    )
+                    getattr(self, f"magnitude_{band}").append(mag)
+            lightcurve_data = {"MJD": time}
+            for band in self.band_list:
+                lightcurve_data["ps_mag_" + band] = getattr(self, f"magnitude_{band}")
+        else:
+            lightcurve_data = {}
```

Comment:
do you store at least the inputs that go into RandomizedSupernova() such that when we actually need it down the road, that it can be generated?

Reply:
The only numerical input in the RandomizedSupernova() class is redshift and this is stored. Other inputs are all strings and are available as self variable through LensPop class. These are population level parameters.